start/sw/web-app/about.md

```diff
@@ -1,10 +1,31 @@
+
+That all mentioned so far, the front-end, is just one part. Another part is a back-end, that is usually written in dynamic languages like [Python]({% link tech/languages/python/flask-installation.md %}), [Ruby]({% link tech/languages/ruby/ruby-installation.md %}), [PHP]({% link tech/languages/php/php-installation.md %}), [JavaScript]({% link tech/languages/nodejs/nodejs.md %}), or some new, even compiled ones like [Golang]({% link tech/languages/go/go-installation.md %}) or [Rust]({% link tech/languages/rust/rust-installation.md %}). In simple scenarios, the static HTML pages might be only served by a web server ([Apache]({% link start/sw/web-app/apache.md %})) or [Nginx]({% link start/sw/web-app/nginx.md %})). As an opposite, most complex applications usually work with a database, on top of the code of the application itself. The most favourite open-source relational databases are [PostgreSQL]({% link tech/database/postgresql/about.md %}) or [MariaDB]({% link tech/database/mariadb/about.md %}).
+
+A client application is most often a browser like Firefox or Chromium, which talks to the server via HTTP protocol. It does not need to be that case always, a client application might be also substituted by a [GUI]({% link start/sw/gui-app/about.md %}) or [terminal]({% link start/sw/cli-app/about.md %}) application.
+
+## Certificates
+
+No matter which technology you use on the server-side, what is common for the web applications is that the trafic between client and server machine goes over the Internet. To not reveal any sensitive information, many sites have utilized an SSL-secured connection over HTTP, shortly HTTPS. To work properly, the server needs to have a valid certificate, which either is obtained and signed by one of the authorities, or can be generated using service [Let's Encrypt](https://letsencrypt.org/). Such certificates require management though, so read about how to secure your webserver with improved Certbot in the [Fedora magazine](https://fedoramagazine.org/secure-your-webserver-improved-certbot/).
+
+## Editing the source-code
+
+For writing code in such heterogeneous environment, one needs a good IDE editor. Wes, `vim` works as great, but there are some alternatives available in the Fedora OS as well.
+
+[PyCharm](https://www.jetbrains.com/pycharm/), a very popular IDE for Python development, is available as a [copr repository](https://copr.fedorainfracloud.org/coprs/phracek/PyCharm/). Microsoft's famous Visual Studio, rewritten to a modern web-based platform based on open-source technologies, is called [Visual Studio Code](https://code.visualstudio.com/) -- [read here](https://fedoramagazine.org/using-visual-studio-code-fedora/) how you can install it on your Fedora. [Atom](https://atom.io/) is another modern IDE, which can be [installed easily](https://fedoramagazine.org/install-atom-fedora/) as well.
```

Comment:
Instead of explaining what it is, could we actually point to specific pages- in python, C#, ... section(I hope it exists)? It would be great to have as less duplicate text as possible). Please also make it a list (or a table, like on the C# page.).

Reply:
Converted to list.